include/envoy/network/listener.h

```diff
@@ -90,6 +95,12 @@ class ListenerConfig {
    * @return const std::string& the listener's name.
    */
   virtual const std::string& name() const PURE;
+
+  /**
+   * @return factory pointer if listening on UDP socket, otherwise return
+   * nullptr.
+   */
+  virtual const Server::ActiveUdpListenerFactory* udpListenerFactory() PURE;
```

Comment:
Sorry, I just took another pass through this and can we s/ActiveUdpListenerFactory/UdpListenerFactory? I think we can lose the active at the interface layer. This naming change should bubble everywhere except for inside connection_handler_impl (where TBH I don't remember why I named everything starting w/ "Active", but probably to connote that they are things currently being served - given the class proliferation if that naming bothers folks we can change it, but let's do that in a different PR).

Reply:
We have Network::UdpListener along aside ActiveUdpListener which is a UdpListenerCallbacks. Would the name become more confusing after removing the prefix "Active"?

Reply:
Sorry can you link me to what you are referring to?

Reply:
We have UdpListener (https://github.com/envoyproxy/envoy/blob/975c62dd2258fe23c3d8098b99261fffd43da736/include/envoy/network/listener.h#L210) and UdpListenerCallbacks (https://github.com/envoyproxy/envoy/blob/975c62dd2258fe23c3d8098b99261fffd43da736/include/envoy/network/listener.h#L155).
The current ActiveUdpListener is an implementation of UdpListenerCallbacks, so would ActiveQuicListener be. So I think their factory class names shouldn't mix with the UdpListener.

Reply:
`UdpListenerFactory` seems fine to me in this context in well paralleled to the listener itself and the callbacks? I'm probably just not understanding your concern though...